HerPublicWebsite/Controllers/QuestionnaireController.cs

```diff
@@ -57,127 +57,127 @@ public IActionResult Index()
 
     [HttpGet("boiler")]
     [ExcludeFromSessionExpiry]
-    public IActionResult GasBoiler_Get()
+    public IActionResult GasBoiler_Get(QuestionFlowStep? entryPoint)
     {
         var questionnaire = questionnaireService.GetQuestionnaire();
         var viewModel = new GasBoilerViewModel
         {
             HasGasBoiler = questionnaire.HasGasBoiler,
-            BackLink = GetBackUrl(QuestionFlowStep.GasBoiler, questionnaire)
+            BackLink = GetBackUrl(QuestionFlowStep.GasBoiler, questionnaire, entryPoint)
         };
 
         return View("GasBoiler", viewModel);
     }
 
     [HttpPost("boiler")]
     [ExcludeFromSessionExpiry]
-    public IActionResult GasBoiler_Post(GasBoilerViewModel viewModel)
+    public IActionResult GasBoiler_Post(GasBoilerViewModel viewModel, QuestionFlowStep? entryPoint)

```

Comment:
On the post methods you don't need the entryPoint parameter - it's already on all of the view models